*** basic-v222p5-vasm.asm	Tue Apr 25 02:53:45 2023
--- basic.asm	Tue Apr 25 03:21:40 2023
***************
*** 1,5 ****
  
! ; Enhanced BASIC to assemble under 6502 simulator, $ver 2.22p5
  
  ; $E7E1 $E7CF $E7C6 $E7D3 $E7D1 $E7D5 $E7CF $E81E $E825
  
--- 1,5 ----
  
! ; Enhanced BASIC to assemble under 6502 simulator, $ver 2.30 (2.22p5-7)
  
  ; $E7E1 $E7CF $E7C6 $E7D3 $E7D1 $E7D5 $E7CF $E81E $E825
  
***************
*** 40,45 ****
--- 40,47 ----
  ;      5.5     garbage collection may cause an overlap with temporary strings
  ;      5.6     floating point multiply rounding bug
  ;      5.7     VAL() may cause string variables to be trashed
+ ;
+ ; 2.30      restarted versioning
  
  ; zero page use ..
  
***************
*** 361,370 ****
  TK_BITCLR         = TK_BITSET+1     ; BITCLR token
  TK_IRQ            = TK_BITCLR+1     ; IRQ token
  TK_NMI            = TK_IRQ+1        ; NMI token
  
  ; secondary command tokens, can't start a statement
  
! TK_TAB            = TK_NMI+1        ; TAB token
  TK_ELSE           = TK_TAB+1        ; ELSE token
  TK_TO             = TK_ELSE+1       ; TO token
  TK_FN             = TK_TO+1         ; FN token
--- 363,373 ----
  TK_BITCLR         = TK_BITSET+1     ; BITCLR token
  TK_IRQ            = TK_BITCLR+1     ; IRQ token
  TK_NMI            = TK_IRQ+1        ; NMI token
+ TK_QUIT           = TK_NMI+1        ; QUIT token
  
  ; secondary command tokens, can't start a statement
  
! TK_TAB            = TK_QUIT+1       ; TAB token
  TK_ELSE           = TK_TAB+1        ; ELSE token
  TK_TO             = TK_ELSE+1       ; TO token
  TK_FN             = TK_TO+1         ; FN token
***************
*** 7994,8003 ****
  
  LAB_MSZM
        .byte $0D,$0A,"Memory size ",$00
- 
  LAB_SMSG
        .byte " Bytes free",$0D,$0A,$0A
!       .byte "Enhanced BASIC 2.22p5",$0A,$00
  
  ; numeric constants and series
  
--- 7997,8010 ----
  
  LAB_MSZM
        .byte $0D,$0A,"Memory size ",$00
  LAB_SMSG
        .byte " Bytes free",$0D,$0A,$0A
! 
! LAB_HELLO
!       .byte "Enhanced BASIC "
! LAB_VERSION
!       .byte "2.30"
!       .byte $0A,$00
  
  ; numeric constants and series
  
***************
*** 8518,8523 ****
--- 8525,8534 ----
  LBB_PRINT
        .byte "RINT",TK_PRINT   ; PRINT
        .byte $00
+ .IF STANDALONE
+ LBB_QUIT
+       .byte "UIT",TK_QUIT     ; QUIT
+ .ENDIF
  TAB_ASCR
  LBB_READ
        .byte "EAD",TK_READ     ; READ
